### PR TITLE
EFRS-704: Handled an error when there is no x-frs-api-key header

### PR DESCRIPTION
**Pull request by andreigaevsky**

### CODE REVIEW

**Review by daffy- — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/exadel/frs/core/trainservice/filter/SecurityValidationFilter.java [hunk 1]
```diff
@@ -86,21 +86,18 @@ public void doFilter(
...
-        val key = apiKey.get(0);
-        try {
-            if (key.length() < 36) {
-                throw new IllegalArgumentException("UUID length is incorrect");
-            }
-            UUID.fromString(key);
-        } catch (Exception e) {
-            val objectResponseEntity = handler.handleBadFormatModelKeyException(new BadFormatModelKeyException());
-            buildException(httpResponse, objectResponseEntity);
+        if (!apiKey.isEmpty()) {
```
Comment (daffy-) [inline api/src/main/java/com/exadel/frs/core/trainservice/filter/SecurityValidationFilter.java:89]: redundant check - we always have an apiKey

File: api/src/main/java/com/exadel/frs/core/trainservice/handler/ResponseExceptionHandler.java [hunk 2]
```diff
@@ -66,4 +71,25 @@
...
+    }
+
+    public ResponseEntity<Object> handleMissingRequestHeader(final String headerName) {
+        return ResponseEntity.status(MISSING_REQUEST_HEADER.getHttpStatus())
+                             .body(new RetrainResponse("Missing header: " + headerName));
+    }
+
+    @ExceptionHandler(ConstraintViolationException.class)
+    ResponseEntity<Object> handleConstraintViolationException(ConstraintViolationException e) {
+        StringBuilder sb = new StringBuilder();
```
Comment (daffy-) [inline api/src/main/java/com/exadel/frs/core/trainservice/handler/ResponseExceptionHandler.java:87]: `val`

File: api/src/main/java/com/exadel/frs/core/trainservice/handler/ResponseExceptionHandler.java [hunk 3]
```diff
@@ -66,4 +71,25 @@
...
+        for(ConstraintViolation violation : e.getConstraintViolations()) {
```
Comment (daffy-) [inline api/src/main/java/com/exadel/frs/core/trainservice/handler/ResponseExceptionHandler.java:88]: `val`

File: api/src/main/java/com/exadel/frs/core/trainservice/validation/UuidValidator.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    @Override
+    public boolean isValid(String uuid, ConstraintValidatorContext cxt) {
+        if (uuid.length() != 36) {
+            return false;
+        }
+        try {
+            UUID.fromString(uuid);
+            return true;
```
Comment (daffy-) [inline api/src/main/java/com/exadel/frs/core/trainservice/validation/UuidValidator.java:21]: add extra empty line before return statement

**Review by daffy- — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/exadel/frs/core/trainservice/filter/SecurityValidationFilter.java (hunk 1 above)
Comment (andreigaevsky) [reply api/src/main/java/com/exadel/frs/core/trainservice/filter/SecurityValidationFilter.java:89]: But if a request doesn't have X_FRS_API_KEY_HEADER, apiKey is empty and apiKey.get(0) has IndexOutOfBoundsException.

**Review by daffy- — APPROVED** [review]

**Pull request merged**